krabcake/rs_hello/src/lib.rs

```diff
@@ -286,24 +274,18 @@ pub extern "C" fn rs_client_request_borrow_mut(
     ret: *mut c_size_t,
 ) -> bool {
     unsafe {
-        vgPlain_dmsg(
-            "lib.rs: handle client request BORROW_MUT 0x%llx\n\0".as_ptr() as *const c_char,
-            *arg.offset(1),
-            ret,
-        );
         COUNTER = COUNTER.next();
         let stash_addr = *arg.offset(1);
         let borrowing_addr = *stash_addr as vg_addr;
         let stash_addr = stash_addr as vg_addr;
-        let lookup = if_addr_has_stack_then(borrowing_addr, |entries| {
-            entries.push(Item::Unique(COUNTER));
-        });
-        if lookup.is_none() {
-            let mut v = Vec::new();
-            v.push(Item::Unique(COUNTER));
-            STACKS.push((borrowing_addr, v));
-        }
+        let stack_idx = STACKS.push(borrowing_addr);
+        let stack_dbg_id = STACKS.0.get(stack_idx).unwrap().dbg_id();
         TRACKED_ADDRS.push((stash_addr, COUNTER));
+        vgPlain_dmsg(
+            "lib.rs: handle client request BORROW_MUT. borrowing_addr: 0x%08llx\n\0".as_ptr() as *const c_char,
+            stack_dbg_id,
+            ret,
+        );
```

Comment:
I changed this to print the ID of `borrowing_addr` but IDK if this makes sense. Was the intention to see the `stack_addr` or the `borrowing_addr` (or both?) @pnkfelix

Reply:
This is just debugging output, its not really intended for end user consumption in the long term. So it doesn't matter to me right now whether it is normalized or not. (I suspect the right thing for debugging output, which I do not expect to be tested by our CI, is to emit *both* the raw underlying address as well as the normalized address-id.)